soco/core.py

```diff
@@ -1578,6 +1580,246 @@ def _build_album_art_full_uri(self, url):
+
+        If a comma formatted string to tracks is specified, then use
+        a similiar string to specify new_pos. Those operations should be
+        ordered from the end of the list to the beginning
+
+        See the helper methods
+        :py:meth:`clear_sonos_playlist`, :py:meth:`move_in_sonos_playlist`,
+        :py:meth:`remove_from_sonos_playlist` for simplified usage.
+
+        update_id - If you have a series of operations, tracking the update_id
+        and setting it, will save a lookup operation.
+
+        Examples:
+          To reorder the first two tracks::
+
+            # sonos_playlist specified by the DidlPlaylistContainer object
+            sonos_playlist = device.get_sonos_playlists()[0]
+            device.reorder_sonos_playlist(sonos_playlist,
+                                          tracks=[0, ], new_pos=[1, ])
+            # OR specified by the item_id
+            device.reorder_sonos_playlist('SQ:0', tracks=[0, ], new_pos=[1, ])
+
+          To delete the second track::
+
+            # tracks/new_pos are a list of int
+            device.reorder_sonos_playlist(sonos_playlist,
+                                          tracks=[1, ], new_pos=[None, ])
+            # OR tracks/new_pos are a list of int-like
+            device.reorder_sonos_playlist(sonos_playlist,
+                                          tracks=['1', ], new_pos=['', ])
+            # OR tracks/new_pos are strings - no transform is done
+            device.reorder_sonos_playlist(sonos_playlist, tracks='1',
+                                          new_pos='')
+
+          To reverse the order of a playlist with 4 items::
+
+            device.reorder_sonos_playlist(sonos_playlist, tracks='3,2,1,0',
+                                          new_pos='0,1,2,3')
+
+        Args:
+            sonos_playlist
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`): The
+                Sonos playlist object or the item_id (str)of the Sonos
+                playlist.
+            tracks: (list): list of track indices(int) to reorder. May also be
+                a list of int like things. i.e. ``['0', '1',]`` OR it may be a
+                str of comma separated int like things. ``"0,1"``.  Tracks are
+                **0**-based. Meaning the first track is track 0, just like
+                indexing into a Python list.
+            new_pos (list): list of new positions (int|None)
+                corresponding to track_list. MUST be the same type as
+                ``tracks``. **0**-based, see tracks above. ``None`` is the
+                indicator to remove the track. If using a list of strings,
+                then a remove is indicated by an empty string.
+            update_id (int): operation id (default: 0) If set to 0, a lookup
+                is done to find the correct value.
+
+        Returns:
+            dict: Which contains 3 elements. change, length and update_id.
```

Comment:
done

On Mon, Sep 14, 2015 at 7:36 AM, Kenneth Nielsen <notifications@github.com>
wrote:

> In soco/core.py
> <https://github.com/SoCo/SoCo/pull/352#discussion_r39387993>:
>
> > +                playlist.
> > +            tracks: (list): list of track indices(int) to reorder. May also be
> > +                a list of int like things. i.e. ``['0', '1',]`` OR it may be a
> > +                str of comma separated int like things. ``"0,1"``.  Tracks are
> > +                **0**-based. Meaning the first track is track 0, just like
> > +                indexing into a Python list.
> > +            new_pos (list): list of new positions (int|None)
> > +                corresponding to track_list. MUST be the same type as
> > +                ``tracks``. **0**-based, see tracks above. ``None`` is the
> > +                indicator to remove the track. If using a list of strings,
> > +                then a remove is indicated by an empty string.
> > +            update_id (int): operation id (default: 0) If set to 0, a lookup
> > +                is done to find the correct value.
> > +
> > +        Returns:
> > +            dict: Which contains 3 elements. change, length and update_id.
>
> "elements" colon, semi-colon?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SoCo/SoCo/pull/352/files#r39387993>.
>



-- 
Best,

Jeff Hinrichs
dundeemt@gmail.com
402.218.1473